src/BeadingStrategy/RedistributeBeadingStrategy.cpp

```diff
@@ -0,0 +1,72 @@
+//Copyright (c) 2020 Ultimaker B.V.
+//CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#include "RedistributeBeadingStrategy.h"
+
+#include <algorithm>
+
+namespace cura
+{
+    BeadingStrategy::Beading RedistributeBeadingStrategy::compute(coord_t thickness, coord_t bead_count) const
```

Comment:
shouldn't `bead_count` be of `size_t`

Reply:
First, `size_t` is a pointer size, it is often used for counts of things, since, as a pointer size, it's also the max of what fits into sequences like vector or list and the like. However, while there is a small number of beads in a vector in the end, the majority of bead count as used in the code is to do math with things that very much _should_ be a `coord_t`, so I'd like to keep it that way.

Reply:
Yes but see commit below ;-)

Reply:
`coord_t` has the same bit size. We specialise `coord_t` to always be `int64_t`. `size_t` is usually the same as `intptr_t` which is 64 bits on all of our builds (since we only ship 64-bit Cura versions now).

Reply:
True, but `coord_t` is signed, and `size_t` is unsigned. In calculations, it can be useful to go below 0, without having to cast, even if the input is a count.

Reply:
In those cases it would be better to use a plain integer then rather than `coord_t`.